ServerCore/Pages/Hints/Edit.cshtml.cs

```diff
@@ -53,6 +53,16 @@ public async Task<IActionResult> OnPostAsync()
             try
             {
                 await _context.SaveChangesAsync();
+
+                var puzzleName = await _context.Hints.Where(m => m.Id == Hint.Id).Select(m => m.Puzzle.Name).FirstOrDefaultAsync();
+
+                var teamMembers = await (from TeamMembers tm in _context.TeamMembers
+                                         join HintStatePerTeam hspt in _context.HintStatePerTeam on tm.Team equals hspt.Team
+                                         where hspt.Hint.Id == Hint.Id
+                                         select tm.Member.Email).ToListAsync();
```

Comment:
I think this needs to check what the state is / whether or not the team used the hint.